### DISCUSSION TITLE
Modify POS => Hardware Station call time-out.

### DISCUSSION THREAD

**Discussion by SasoGLIGOR** [microsoft/Dynamics365Commerce.Solutions#203] [Q&A] [answered]
Hello,

We are doing some custom Hardware Station development regarding a payment connector integration.
In our development there is call from POS to Hardware Station, custom method, which is not instantaneous.
The execution of this method takes about 3 minutes, and the standard timeout on the POS => Hardware Station call is 2 minutes.
I wanted to know if there is any possibility to use a different timeout value for the POS => Hardware Station custom calls.

Thank you.

**Comment by madyke** [comment, marked as answer]
Hi @SasoGLIGOR - unfortunately, I don't believe this is possible. We have the `HardwareStationDeviceActionRequest`, but it does not have a configurable timeout and should just use the default 2 minutes.

If you aren't able to find a workaround in the payment connector for dealing with the default timeout and still require the timeout to be configurable, then you'll have to submit an extensibility request:
[Extensibility requests](https://learn.microsoft.com/en-us/dynamics365/fin-ops-core/dev-itpro/extensibility/extensibility-requests)

**Reply by SasoGLIGOR** [reply]
Hello @madyke,

Thank you for your feedback on the subject.
We have tried modifying the timeout with the HardwareStationDeviceActionRequest object, but it doesn’t work, as you mentioned.
After contacting the MS support, we were informed that there is an issue that could be of help in our case : [MS Issue 828512](https://fix.lcs.dynamics.com/Issue/Details?kb=0&bugId=828512&dbType=3&qc=bc77f1307631d7c6100061b8b32c17ed816f974d28089ecbc83d42ed2c3effb9)
Haven’t had the opportunity to test it because we are not on the required version, but will test it as soon as we update.

Kind regards.
